server/auth_server.go

```diff
@@ -614,19 +614,26 @@ func UpgradeCase() error {
 
 	dbSettings, err := readCommonSettings(settings)
 	if err != nil {
-		log.Errorf("GetConfig: Error reading DB settings %v", err)
+		log.Errorf("UpgradeCase: Error reading DB settings %v", err)
 		return err
 	}
 
 	// Get all provider specific settings from setting table for previously configured auth control
 	for _, val := range providers.Providers {
+		if !providers.IsProviderSupported(val) {
+			log.Debug("Auth provider not supported by rancher-auth-service")
+			return nil
+		}
```

Comment:
providers.Providers is a list in rancher-auth-service itself, right? I dont see the reason why we need this check while iterating that list